core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -380,6 +380,10 @@ private synchronized void initServer() throws IOException {
     RaftServerConfigKeys.Log.Appender.setInstallSnapshotEnabled(
         properties, false);
 
+    // if left enabled, the System.exit() called by Ratis can deadlock with the AlluxioMaster
+    // process shutdown hook
+    org.apache.ratis.util.ExitUtils.disableSystemExit();
```

Comment:
Please update the comment here as well. For this one, it's OK to have a big comment.